projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.integration.spec.ts

```diff
@@ -918,166 +697,47 @@ describe('IgxHierarchicalGrid Integration #hGrid', () => {
 
-            firstHeaderIcon.click();
-            await wait();
+            UIInteractions.clickElement(firstHeaderIcon);
             fixture.detectChanges();
+            tick();
 
-            headers = childGrid.nativeElement.querySelectorAll('igx-grid-header-group');
-            expect(childGrid.columnList.toArray()[0].pinned).toBeTruthy();
-            expect(headers[0].classList.contains('igx-grid__th--pinned')).toBeTruthy();
+            childHeader = GridFunctions.getColumnGroupHeaders(fixture)[4];
+            expect(childGrid.columnList.first.pinned).toBeTruthy();
+            expect(childHeader.nativeElement.classList).toContain('igx-grid__th--pinned');
         }));
 
         it('should be applied correctly for child grid with multi-column header.', (() => {
-            const ri = fixture.componentInstance.rowIsland;
-            const col = ri.columnList.find(x => x.header === 'Information');
-            col.pinned = true;
+            fixture.componentInstance.rowIsland.columnList.find(x => x.header === 'Information').pinned = true;
             fixture.detectChanges();
 
-            const row = hierarchicalGrid.getRowByIndex(0) as IgxHierarchicalRowComponent;
-            UIInteractions.clickElement(row.expander);
-            fixture.detectChanges();
+            hierarchicalGrid.expandRow(hierarchicalGrid.dataRowList.first.rowID);
 
-            const childGrids =  fixture.debugElement.queryAll(By.css('igx-child-grid-row'));
-            const childGrid = childGrids[0].query(By.css('igx-hierarchical-grid')).componentInstance;
+            const childGrid = hierarchicalGrid.hgridAPI.getChildGrids(false)[0];
             // check unpinned/pinned columns
             expect(childGrid.pinnedColumns.length).toBe(3);
             expect(childGrid.unpinnedColumns.length).toBe(1);
-             // check cells
-            const cells = childGrid.getRowByIndex(0).cells;
-            expect(cells.length).toBe(3);
+            // check cells
+            expect(childGrid.getRowByIndex(0).cells.length).toBe(3);
             let cell = childGrid.getCellByColumn(0, 'ChildLevels');
             expect(cell.visibleColumnIndex).toEqual(0);
-            expect(cell.nativeElement.classList.contains('igx-grid__td--pinned')).toBe(true);
+            expect(cell.nativeElement.classList).toContain('igx-grid__td--pinned');
             cell = childGrid.getCellByColumn(0, 'ProductName');
             expect(cell.visibleColumnIndex).toEqual(1);
-            expect(cell.nativeElement.classList.contains('igx-grid__td--pinned')).toBe(true);
+            expect(cell.nativeElement.classList).toContain('igx-grid__td--pinned');
             cell = childGrid.getCellByColumn(0, 'ID');
             expect(cell.visibleColumnIndex).toEqual(2);
-            expect(cell.nativeElement.classList.contains('igx-grid__td--pinned')).toBe(false);
+            expect(cell.nativeElement.classList).not.toContain('igx-grid__td--pinned');
```

Comment:
You can create a constant for the class igx-grid__td--pinned or to add a new method in the GridFunctions to check pinned class.